packages/contracts-bedrock/test/L1/OPContractsManager.t.sol

```diff
@@ -315,9 +316,15 @@ contract OPContractsManager_Upgrade_Harness is CommonTest {
         // then reset its code to the original code.
         bytes memory delegateCallerCode = address(_delegateCaller).code;
         vm.etch(_delegateCaller, vm.getDeployedCode("test/mocks/Callers.sol:DelegateCaller"));
+
         DelegateCaller(_delegateCaller).dcForward(
             address(opcm), abi.encodeCall(IOPContractsManager.upgrade, (opChainConfigs))
         );
+
+        VmSafe.Gas memory gas = vm.lastCallGas();
+
+        assertLt(gas.gasTotalUsed, 15_000_000, "Upgrade exceeds gas target of 15M");
```

Comment:
This way we have some margin to the 15M number in case estimation is wrong
```suggestion
        assertLt(gas.gasTotalUsed, 0.9 * 15_000_000, "Upgrade exceeds gas target of 15M");
```